src/mlpack/methods/cf/decomposition_policies/batch_svd_method.hpp

```diff
@@ -73,6 +69,89 @@ class BatchSVDPolicy
       svdbatch.Apply(cleanedData, rank, w, h);
     }
   }
+
+  /**
+   * Return predicted rating given user ID and item ID.
```

Comment:
If I am not mistaken, this code duplicates six times. I think it is a good idea to add another abstraction layer. How do you think?

Reply:
Hmm I understand what you mean but I am not sure how to add an abstraction layer. If we are going to add another layer (e.g. maybe a wrapper class), we also need to consider how to extend it to `BiasSVD` and `SVD++` which will be implemented.

Reply:
I think that should become clear as soon as `BiasSVD` and `SVD++` are implemented.
Right now I've got the following idea:
Actually, we have 2 policies: `CholevskyPolicy` and `SVDPolicy`. Meanwhile, `CholevskyPolicy` depends on matrices `W` and `H`, so it has another template argument which can be equal to `BatchSVDPolicy`, `NMFPolicy`, `RandomizedSVDPolicy` and so on.
For example:
```
template<typename DecompositionPolicy>
class CholevskyNeighborhoodPolicy
{
  DecompositionPolicy p;

  void Apply(.......)
  {
    p.Apply(............);
  }

  double GetRating(..........);
  void GetNeighbourhood(.........);
};
```
Thus, let's wait for `BiasSVD` and `SVD++`, then maybe I'll suggest something else.

Reply:
Okay! Many thanks!